pyspecProcScripts/correlation_alignment.py

```diff
@@ -52,8 +61,23 @@ def correl_align(s, align_phases=False,tol=1e-4,indirect_dim='indirect',fig_titl
                 the width of the Gaussian function used to frequency filter
                 the data in the calculation of the correlation function.
     """
-    ph1_len = len(s.getaxis('ph1'))
-    ph2_len = len(s.getaxis('ph2'))
+    if 'nScans' in s.dimlabels:
+        phcycdims = [j for j in s.dimlabels if j.startswith('ph')]
+        indirect = set(s.dimlabels)-set(phcycdims)-set([direct])
+        indirect = [j for j in s.dimlabels if j in indirect]
+        nScans_len = len(s.getaxis('nScans'))
+        s.smoosh(indirect)
+    if 'ph2' in s.dimlabels:
+        ODNP=False
+    else:
+        ODNP=True
+    if ODNP:
+        s.ift(['ph1'])
+        ph1_len = len(s.getaxis('ph1'))
+    else:
+        s.ift(['ph1','ph2'])
+        ph1_len = len(s.getaxis('ph1'))
+        ph2_len = len(s.getaxis('ph2'))
```

Comment:
if you have a dictionary, the ift would be a loop over keys, and the lengths could be constructed with a list comprehension

Reply:
done I think